Source/InvocationCollection.cs

```diff
@@ -69,7 +69,13 @@ public int Count
 		{
 			get
 			{
-				lock (this.invocations)
+				// Test 
+				if (this.invocations == null)
+				{
+					throw new IndexOutOfRangeException();
+				}
```

Comment:
Unfortunately, this still isn't sufficient. The array can have a capacity that exceeds 'count' so you are not going to get an out-of-range exception in some cases. I suggest you just add a straightforward check for `0 <= index && index < this.count` instead. ;-)